lib/krb5/init_creds_pw.c

```diff
@@ -2524,18 +2968,35 @@ init_creds_step(krb5_context context,
 		krb5_krbhst_info *hostinfo,
 		unsigned int *flags)
 {
+    struct timeval start_time, end_time;
+    krb5_data checksum_data;
     krb5_error_code ret;
     size_t len = 0;
     size_t size;
     AS_REQ req2;
 
+    gettimeofday(&start_time, NULL);
+
     krb5_data_zero(out);
+    krb5_data_zero(&checksum_data);
 
     if (ctx->as_req.req_body.cname == NULL) {
 	ret = init_as_req(context, ctx->flags, &ctx->cred,
 			  ctx->addrs, ctx->etypes, &ctx->as_req);
-	if (ret)
+	if (ret) {
+	    free_init_creds_ctx(context, ctx);
```

Comment:
This looks wrong.  None of the other error paths in this function call free_init_creds_ctx().